.circleci/config.yml

```diff
@@ -55,13 +64,23 @@ commands:
           # If a new commit is built, it will fall back on the most recent cache from the same branch.
           key: sccache-cache-{{ .Environment.CIRCLE_JOB }}-{{ .Branch }}-{{ .Revision }}
           paths:
-            - "~/.cache/sccache"
+            - ~/.cache/sccache
+  # This is only performed by the workspace clippy job, since it does the largest cargo fetch.
+  # Other jobs might restore a slightly older copy of this cache.
+  save-cargo-cache:
+    steps:
+      # Discard crates.io patches so that it produces a re-usable cache key checksum.
+      - run:
+          name: Restore Cargo.lock
+          command: git restore Cargo.lock
```

Comment:
Will this cause failures when we have unreleased versions of our crates running in CI?

Reply:
This is only for ensuring proper caching in CI. Should never disturb.